c/tskit/tables.c

```diff
@@ -1329,39 +1329,76 @@ tsk_individual_table_load(tsk_individual_table_t *self, kastore_t *store)
  *************************/
 
 static int
-tsk_node_table_expand_main_columns(tsk_node_table_t *self, tsk_size_t additional_rows)
+calculate_max_rows(tsk_size_t num_rows, tsk_size_t max_rows,
+    tsk_size_t max_rows_increment, tsk_size_t additional_rows, tsk_size_t *new_max_rows)
 {
     int ret = 0;
-    tsk_size_t increment = TSK_MAX(additional_rows, self->max_rows_increment);
-    tsk_size_t new_size = self->max_rows + increment;
 
-    if (check_table_overflow(self->max_rows, increment)) {
+    if (check_table_overflow(num_rows, additional_rows)) {
         ret = TSK_ERR_TABLE_OVERFLOW;
         goto out;
     }
-    if ((self->num_rows + additional_rows) > self->max_rows) {
-        ret = expand_column((void **) &self->flags, new_size, sizeof(tsk_flags_t));
+
+    if (num_rows + additional_rows <= max_rows) {
+        *new_max_rows = max_rows;
+        goto out;
+    }
+
+    if (max_rows_increment == 0) {
+        /* Doubling by default */
+        *new_max_rows = TSK_MIN(max_rows * 2, TSK_MAX_ID + (tsk_size_t) 1);
```

Comment:
There's a lot of pointer deferencing here with ``*new_max_rows`` - maybe call the parameter ``ret_new_max_rows``, and a local vaiable ``new_max_rows`` and assign to the return pointer at the end of the function?

That would make the control flow a bit clearer as well I think.

Reply:
Fixed in https://github.com/tskit-dev/tskit/pull/1683/commits/c3db90eb1ce8f3a4927c24d7dd06e4ec0c0a9bef